imap_processing/idex/IDEXPacketParser.py

```diff
@@ -0,0 +1,120 @@
+                    if evtnum not in self.data[scitype]:
+                        self.data[scitype][evtnum] = pkt.data['IDX__SCI0RAW'].raw_value
+                    else:
+                        self.data[scitype][evtnum] += pkt.data['IDX__SCI0RAW'].raw_value
+
+        # Parse the waveforms according to the scitype present (high gain and low gain channels encode waveform data differently).
+        self.scitype_to_names = {2: "TOF_High", 4: "TOF_Low", 8: "TOF_Mid", 16: "Target_Low",
+                                32: "Target_High", 64: "Ion_Grid"}
+        datastore = {}
+        self.time_low_sr = []
+        self.time_high_sr = []
+        for scitype in self.data:
+            datastore[self.scitype_to_names[scitype]] = []
+            for evt in self.data[scitype]:
+                datastore[self.scitype_to_names[scitype]].append(self._parse_waveform_data(self.data[scitype][evt], scitype))
+                if self.scitype_to_names[scitype] == 'Target_Low':
+                    self.time_low_sr.append(np.linspace(0, len(datastore['Target_Low'][0]), len(datastore['Target_Low'][0])))
+                if self.scitype_to_names[scitype] == 'TOF_Low':
+                    self.time_high_sr.append(np.linspace(0, len(datastore['TOF_Low'][0]), len(datastore['TOF_Low'][0])))
+
+        self.data = datastore
+        self.numevents = evtnum
+    
+    def _log_packet_info(self, evtnum, pkt):
+        print(f"^*****Event header {evtnum}******^")
+        # Extract the 17-22-bit integer (usually 8)
+        self.lspretrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value >> 16) &  0b1111
+        # Extract the next 4-bit integer (usually 8)
+        self.lsposttrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value >> 12) & 0b1111
+        # Extract the next 6 bits integer (usually 32)
+        self.hspretrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value >> 6) & 0b111111
+        # Extract the first 6 bits (usually 32)
+        self.hsposttrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value) & 0b111111
+        print("HS pre trig sampling blocks: ", self.hspretrigblocks)
+        print("LS pre trig sampling blocks: ", self.lspretrigblocks)
+        print("HS post trig sampling blocks: ", self.hsposttrigblocks)
+        print("LS post trig sampling blocks: ", self.lsposttrigblocks)
+        self.TOFdelay = pkt.data['IDX__TXHDRSAMPDELAY'].raw_value >> 2  # First two bits are padding
+        mask = 0b1111111111
+        self.lgdelay = (self.TOFdelay) & mask
+        self.mgdelay = (self.TOFdelay >> 10) & mask
+        self.hgdelay = (self.TOFdelay >> 20) & mask
+        print(f"High gain delay = {self.hgdelay} samples.")
+        print(f"Mid gain delay = {self.mgdelay} samples.")
+        print(f"Low gain delay = {self.lgdelay} samples.")
+        if(pkt.data['IDX__TXHDRLSTRIGMODE'].derived_value!=0):  # If this was a LS (Target Low Gain) trigger
+            self.Triggerorigin = 'LS' 
```

Comment:
What will these Trigger variables be used for? In the case that some of these `if` conditions are not satisfied, they might not be defined when something tries to access them from the object, so should be initialized under init.

Reply:
yeah these are only used for logging really, so I don't know if it matters that they are like this.  I'm going to get rid of the "self." because I don't think the rest of the class needs to know or care about them.